src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6849,6 +6849,9 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_CollectionBuilderAttributeInvalidMethodName" xml:space="preserve">
     <value>The CollectionBuilderAttribute method name is invalid.</value>
   </data>
+  <data name="ERR_CollectionExpressionEscape" xml:space="preserve">
+    <value>A collection expression of type '{0}' cannot be used in this context because it may be exposed outside of the current scope.</value>
```

Comment:
The more I think about it the more I feel this is a good place to land. It would be nice if we could expand the diagnostic message to suggest casting to array type.